src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -55,7 +57,9 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
             this.tags.addAll(tags);
         }
         if (grades != null) {
-            this.grades.addAll(grades);
+            this.grades = grades;
```

Comment:
does it matter it if we pass a shallow copy? is a deep copy better? (defensive programming)

Reply:
Lol idk man

Reply:
I mean we could do a new HashMap<grades>